### PR TITLE
add switch to GUI and implement alt hold toggle mode

### PR DESCRIPTION
**Pull request by jsschell**
alt hold hold can be activated by using the new button in the main view. By default toggle mode is disabled and crazyflie is in alt hold mode as long as at least one joystick is controlled by the user. On release it will fall immediately.

With toggle mode enabled, there is no need to keep a finger on the screen. For safety reasons there is a big emergency button between the joysticks that should be easy to hit without looking onto the screen. There are three additional settings in [settings -> flight control settings -> alt hold toggle preferences] to allow easy transition from alt hold to normal flying without landing:

- crazyflie will remain hovering for **transition time [s]** after turning off altitude hold via button.
- during this time crazyflie may drop with set **transition drop speed [m/s]**
- and the user may take back thrust control by raising thrust over **transition thrust take-over threshold[%]**.

### CODE REVIEW

**Comment by krichardsson** [comment]
@fredg02 As far as I understand this PR was mainly intended to show possibilities of how to implement altitude hold. It has been around for a long time, what to do?

**Comment by fredg02** [comment]
Thanks @jsschell for this puill request and sorry for the lack of response. I've tested the changes a while ago and I like the idea that at least one finger needs to be on a joystick. The big emergency button is not safe enough IMHO.
TL;DR: I won't be merging this PR, but I might incorporate some of the ideas in an upcoming version. So, closing for now.
